src/Arcus.Testing.Assert/AssertXml.cs

```diff
@@ -370,15 +424,18 @@ private static Node CreateNode(XmlNode xml, string path, AssertXmlOptions option
                    .Where(n => !options.IgnoredNodeNames.Contains(n.LocalName))
                    .ToArray();
 
-            bool hasSingleChild = children.Length == 1;
             return new Node
             {
                 Current = xml,
                 Path = path,
-                Children = children.Select((ch, i) =>
+                Children = children.GroupBy(ch => ch.LocalName).SelectMany(group =>
```

Comment:
Changed this a bit, as the generated XPath for the XML nodes was using the indexer `[]` even though there was only a single element with the given name. By grouping them by name, we only use the indexer in case if there are multiple childeren with the same name (like it should be).